fuse_models/src/odometry_2d_publisher.cpp

```diff
@@ -254,21 +262,114 @@ void Odometry2DPublisher::tfPublishTimerCallback(const ros::TimerEvent& event)
     tf2_2d::Vector2 velocity_linear;
     tf2::fromMsg(odom_output_.twist.twist.linear, velocity_linear);
 
-    tf2_2d::Vector2 unused_acc;
-    double unused_yaw_vel;
-
-    predict(
-      pose,
-      velocity_linear,
-      odom_output_.twist.twist.angular.z,
-      unused_acc,
-      event.current_real.toSec() - odom_output_.header.stamp.toSec(),
-      pose,
-      velocity_linear,
-      unused_yaw_vel,
-      unused_acc);
-
-    trans.header.stamp = event.current_real;
+    const double dt = event.current_real.toSec() - odom_output_.header.stamp.toSec();
+
+    if (params_.predict_odom)
```

Comment:
If we're going for feature parity, that would make sense.

Reply:
Yep, that would make the params easier to understand. If you both agree, I can do the following:
1. Remove the odometry message publication from the `notifyCallback`
2. Remove the `predict_odom` param
3. Publish the TF and odometry message from `tfPublishTimerCallback` only, and always
4. Since both the TF and odometry messsage are published from the timer callback, rename it to `publishTimerCallback`, and I could also rename the other related variables, and even the `tf_publish_frequency` parameter
5. If `predict_to_current_time` is enabled, predict both the TF and the odometry message

BTW I was thinking of also publishing the accelerations in another topic. That could be part of another PR, but it'd be certainly easier if we do simplify things as explained above.

Reply:
That all sounds good to me. And I'd put the accel publish in a separate PR, if you don't mind.

Reply:
Done.

- [x] Remove the odometry message publication from the notifyCallback
- [x] Remove the predict_odom param
- [x] Publish the TF and odometry message from tfPublishTimerCallback only, and always
- [x] Since both the TF and odometry messsage are published from the timer callback, rename it to `publishTimerCallback` and also the timer attribute to `publish_timer_`.
- [x] Also rename the `tf_publish_frequency` parameter to `publish_frequency`. I did this on a separate commit in case we don't want that.
- [x] If predict_to_current_time is enabled, predict both the TF and the odometry message
- [x] :exclamation: I also notice the `publish_tf` parameter wasn't used, so I 've disable broadcasting the TF if it's `false`.

The PR should be reviewed again to confirm all this is good. It should be easier to review now, indeed. I still need to merge the `predict` function and use it as a `ceres::CostFunction`, but I'll have to leave that for next week. I also need to test now that everything still works as intended.